src/hooks/useAuthorites.js

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react'
+import { useDataQuery } from '@dhis2/app-service-data'
+
+const query = {
+    me: {
+        resource: 'me',
+    },
+}
+
+const useAuthorites = (authorities, defaultValue = false) => {
+    const { loading, error, data } = useDataQuery(query)
+
+    const [hasAnyAuthority, setHasAnyAuthority] = useState(defaultValue)
+    const [hasAllAuthorities, setHasAllAuthorities] = useState(defaultValue)
+
+    useEffect(() => {
+        if (!loading) {
+            if (data && data.me && data.me.authorities) {
+                const filter = data.me.authorities.filter(auth =>
+                    authorities.includes(auth)
+                )
+                setHasAnyAuthority(filter.length > 0)
+                setHasAllAuthorities(filter.length === authorities.length)
+            }
+        }
+    }, [loading])
+
+    return { loading, error, hasAnyAuthority, hasAllAuthorities }
```

Comment:
I think I would quite like to just use the approach below. I don't think the use of state and effect is needed at all:
```javascript
// useAppDataQuery.js
const { loading, error, data } = useDataQuery(query)

if (!data) {
    return { loading, error, data }
}

return {
    loading,
    error,
    data: {
        userAuthorities: data.me.authorities,
    }
}
```

So this would then be our primary query hook, which is specifically to "feed into the provider". So we call this in App.js and handle loading and error there first. Once data comes through we set that as a value to the provider.

And then, we can also implement a separate `useAuthorities` hook. This hook can only be used in components that are descendants of the AppContextProvider. This could look a bit like this:
```javascript
// useAuthorities.js
const useAuthorities = authorities => {
    const { userAuthorities } = useAppContext()
    const filter = userAuthorities.filter(auth =>
        authorities.includes(auth)
    )

    return {
        hasAnyAuthority: filter.length > 0,
        hasAllAuthorities: filter.length === authorities.length,
    }
}
```